mlir/include/mlir/Dialect/AMX/AMX.td

```diff
@@ -151,6 +151,8 @@ def TileZeroOp : AMX_Op<"tile_zero", [
     Zeroes the destination tile, with the shape defined by the 2-dim
     vector type of the result. This is eventually lowered into the
     "tilezero" instruction with the corresponding tile configuration.
+    It includes memory effects and CSE doesn't eliminate multiple
+    "tilezero" instructions.
```

Comment:
While it is true, let's focus on the intended semantics and not the implementation behind it.

For example:
```
Each `tile_zero` operation serves as a compilation hint to use a separate tile register.
```
Sth to capture the idea behind the side-effect modeling, and how one should use or what to expect out of the op.